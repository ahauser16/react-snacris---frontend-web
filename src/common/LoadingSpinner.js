import React from "react";
import "./LoadingSpinner.css";

/** Loading message used by components that fetch API data. */

function LoadingSpinner() {
  return (
    <div className="LoadingSpinner" data-testid="loading-spinner">
      Loading...
    </div>
  );
}

export default LoadingSpinner;
