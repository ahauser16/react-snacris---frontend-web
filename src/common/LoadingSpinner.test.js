import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import LoadingSpinner from "./LoadingSpinner";

describe("LoadingSpinner", () => {
  test("renders loading spinner with correct text", () => {
    render(<LoadingSpinner />);

    // Check that the loading text is displayed
    expect(screen.getByText("Loading...")).toBeInTheDocument();
  });

  test("renders with correct CSS class", () => {
    render(<LoadingSpinner />);

    // Check that the component has the correct CSS class
    const spinner = screen.getByTestId("loading-spinner");
    expect(spinner).toHaveClass("LoadingSpinner");
  });

  test("has correct data-testid attribute", () => {
    render(<LoadingSpinner />);

    // Verify the data-testid is present
    const spinner = screen.getByTestId("loading-spinner");
    expect(spinner).toBeInTheDocument();
  });

  test("renders as a div element", () => {
    render(<LoadingSpinner />);

    // Check that it's rendered as a div
    const spinner = screen.getByTestId("loading-spinner");
    expect(spinner.tagName).toBe("DIV");
  });

  test("accessibility - has appropriate loading indication", () => {
    render(<LoadingSpinner />);

    // Check that screen readers can identify loading content
    const spinner = screen.getByTestId("loading-spinner");
    expect(spinner).toHaveTextContent("Loading...");

    // Verify it's accessible by text content
    expect(screen.getByText("Loading...")).toBeInTheDocument();
  });

  test("can be found by role (generic)", () => {
    render(<LoadingSpinner />);

    // Since it's a div, it should have generic role
    const spinner = screen.getByTestId("loading-spinner");
    expect(spinner).toBeInTheDocument();
  });

  test("snapshot test - component structure remains consistent", () => {
    const { container } = render(<LoadingSpinner />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("multiple instances can be rendered", () => {
    render(
      <div>
        <LoadingSpinner />
        <LoadingSpinner />
      </div>
    );

    // Both instances should be present
    const spinners = screen.getAllByTestId("loading-spinner");
    expect(spinners).toHaveLength(2);

    // Both should have the loading text
    const loadingTexts = screen.getAllByText("Loading...");
    expect(loadingTexts).toHaveLength(2);
  });

  test("component is pure - no props needed", () => {
    // Should render without any props
    expect(() => render(<LoadingSpinner />)).not.toThrow();

    // Should render the same output every time
    const { container: firstContainer } = render(<LoadingSpinner />);
    const firstRender = firstContainer.textContent;

    const { container: secondContainer } = render(<LoadingSpinner />);
    const secondRender = secondContainer.textContent;

    expect(firstRender).toBe(secondRender);
    expect(firstRender).toBe("Loading...");
  });

  test("integrates well with parent components", () => {
    const ParentComponent = () => (
      <div data-testid="parent">
        <h1>Data Loading</h1>
        <LoadingSpinner />
      </div>
    );

    render(<ParentComponent />);

    // Parent should contain the spinner
    const parent = screen.getByTestId("parent");
    const spinner = screen.getByTestId("loading-spinner");

    expect(parent).toContainElement(spinner);
    expect(screen.getByText("Data Loading")).toBeInTheDocument();
    expect(screen.getByText("Loading...")).toBeInTheDocument();
  });
});
