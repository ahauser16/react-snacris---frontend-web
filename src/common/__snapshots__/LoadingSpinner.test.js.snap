// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner snapshot test - component structure remains consistent 1`] = `
<div
  class="LoadingSpinner"
  data-testid="loading-spinner"
>
  Loading...
</div>
`;
